tests/test_archs.py

```diff
@@ -0,0 +1,247 @@
+from pythonforandroid.util import BuildInterruptingException
+from pythonforandroid.archs import (
+    Arch,
+    ArchARM,
+    ArchARMv7_a,
+    ArchAarch_64,
+    Archx86,
+    Archx86_64,
+)
+
+expected_env_gcc_keys = {
+    "CFLAGS",
+    "LDFLAGS",
+    "CXXFLAGS",
+    "TOOLCHAIN_PREFIX",
+    "TOOLCHAIN_VERSION",
+    "CC",
+    "CXX",
+    "AR",
+    "RANLIB",
+    "LD",
+    "LDSHARED",
+    "STRIP",
+    "MAKE",
+    "READELF",
+    "NM",
+    "BUILDLIB_PATH",
+    "PATH",
+    "ARCH",
+    "NDK_API",
+}
+
+
+class TestArch(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context()
+        self.ctx.ndk_api = 21
+        self.ctx.android_api = 27
+        self.ctx._sdk_dir = "/opt/android/android-sdk"
+        self.ctx._ndk_dir = "/opt/android/android-ndk"
+        self.ctx.setup_dirs(os.getcwd())
+        self.ctx.bootstrap = Bootstrap().get_bootstrap("sdl2", self.ctx)
+        self.ctx.bootstrap.distribution = Distribution.get_distribution(
+            self.ctx, name="sdl2", recipes=["python3", "kivy"]
+        )
+        self.ctx.python_recipe = Recipe.get_recipe("python3", self.ctx)
+
+    def test_arch(self):
+        arch = Arch(self.ctx)
+        with self.assertRaises(AttributeError) as e1:
+            arch.__str__()
+        self.assertEqual(
+            e1.exception.args[0], "'Arch' object has no attribute 'arch'"
+        )
+        with self.assertRaises(AttributeError) as e2:
+            getattr(arch, "target")
+        self.assertEqual(
+            e2.exception.args[0], "'NoneType' object has no attribute 'split'"
+        )
+        self.assertIsNone(arch.toolchain_prefix)
+        self.assertIsNone(arch.command_prefix)
+        self.assertIsInstance(arch.include_dirs, list)
+
+    # Here we mock two functions:
+    # - `ensure_dir` because we don't want to create any directory
+    # - `find_executable` because otherwise we will
+    #   get an error when trying to find the compiler (we are setting some fake
+    #   paths for our android sdk and ndk so probably will not exist)
+    @mock.patch("pythonforandroid.archs.find_executable")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    def test_arch_arm(self, mock_ensure_dir, mock_find_executable):
+        mock_find_executable.return_value = "arm-linux-androideabi-gcc"
+        mock_ensure_dir.return_value = True
+
+        arch = ArchARM(self.ctx)
+        self.assertEqual(arch.arch, "armeabi")
+        self.assertEqual(arch.__str__(), "armeabi")
+        self.assertEqual(arch.toolchain_prefix, "arm-linux-androideabi")
+        self.assertEqual(arch.command_prefix, "arm-linux-androideabi")
+        self.assertEqual(arch.target, "armv7a-none-linux-androideabi")
+        self.assertEqual(arch.platform_dir, "arch-arm")
+        arch = ArchARM(self.ctx)
+
+        # Check environment flags
+        env = arch.get_env()
+        self.assertIsInstance(env, dict)
+
+        for k in expected_env_gcc_keys:
+            self.assertIn(k, env)
```

Comment:
Minor suggestion, checking the intersection and equality should also work I think:
```python
assert expected_env_gcc_keys == env.keys() & expected_env_gcc_keys
```

Reply:
Good point, it works, but I would rather prefer to use this:

`self.assertEqual(expected_env_gcc_keys, env.keys() & expected_env_gcc_keys)`

Mostly because this way we have a more clear message in case of error

Reply:
Sure feel free :+1: